ui/StatusQ/src/permissionutilsinternal.cpp

```diff
@@ -161,35 +161,37 @@ int /*PermissionTypes::Type*/ PermissionUtilsInternal::isEligibleToJoinAs(QAbstr
     }
 
     QSet<PermissionTypes::Type> tmpRes;
-    bool hasAnyJoinPermission{false};
+    bool hasMemberPermission{false};
     constexpr auto isJoinTypePermission = [](PermissionTypes::Type type) {
         return type == PermissionTypes::Type::TokenMaster ||
                type == PermissionTypes::Type::Admin ||
                type == PermissionTypes::Type::Member;
     };
 
+    constexpr auto isMemberPermission = [](PermissionTypes::Type type) {
+        return type == PermissionTypes::Type::Member;
+    };
+
     const auto permissionsCount = permissionsModel->rowCount();
     for (int i = 0; i < permissionsCount; i++) {
         const auto permissionType = static_cast<PermissionTypes::Type>(permissionsModel->data(permissionsModel->index(i, 0), permissionTypeRole).toInt());
         if (isJoinTypePermission(permissionType)) {
-            hasAnyJoinPermission = true;
+            if (isMemberPermission(permissionType))
+                hasMemberPermission = true;
```

Comment:
`isMemberPermission` is used in a single place. Could be simplified with this

```suggestion
                hasMemberPermission = permissionType == PermissionTypes::Type::Member;
```